lib/src/platform/src/platform.dart

```diff
@@ -21,27 +21,29 @@ class Platform {
   static final StreamsChannel streamsChannel =
       StreamsChannel('io.ably.flutter.stream', codec);
 
-  /// Initializing ably on platform side by invoking `register` platform method.
-  /// Register will clear any stale instances on platform.
-  static Future? _initializer;
+  /// This field will reset to true when the state is reset. This allows us to
+  /// if an app has been restarted, or a hot restart (not hot reload) has
+  /// happened.
+  static bool _isStateReset = true;
 
-  static Future _initialize() async {
-    if (_initializer == null) {
+  /// Clears instances on the Platform side
+  static Future<void> _handleNewState() async {
```

Comment:
The name `_handleNewState` doesn't fit well in here. However, not very important as it is a private method.

Reply:
I tried to pick a name which means both the app is just launching, but also the hot restart is called. Perhaps I could call it `isStateReset`.

Reply:
In 4dce9d58a0d5fffd8f936e38efb0e758337642cc, I have renamed `_isStateReset` to `_shouldClearPlatformInstances`. And `_handleNewState` to `ensureFreshState`. What do you think?

Reply:
It looks like @tiholic's opening comment for this conversation has been addressed so I'm going to resolve this on his behalf.